controllers/mirrorpeersecret_controller.go

```diff
@@ -0,0 +1,176 @@
+	return isObjectASecretWithProvidedLabel(obj, SecretLabelTypeKey, string(DestinationLabel))
+}
+
+// createInternalSecret a common function to create any type secret
+func createInternalSecret(secretNameAndNamespace types.NamespacedName,
+	storageClusterNameAndNamespace types.NamespacedName,
+	secretType SecretLabelType,
+	secretData []byte) *corev1.Secret {
+	retSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretNameAndNamespace.Name,
+			Namespace: secretNameAndNamespace.Namespace,
+			Labels: map[string]string{
+				SecretLabelTypeKey: string(secretType),
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"secretData": secretData,
+		},
+		StringData: map[string]string{
+			"namespace":            storageClusterNameAndNamespace.Namespace,
+			"storage-cluster-name": storageClusterNameAndNamespace.Name,
+		},
+	}
+	return retSecret
+}
+
+// CreateSourceSecret creates a BLUE secret
+func CreateSourceSecret(secretNameAndNamespace types.NamespacedName,
+	storageClusterNameAndNamespace types.NamespacedName,
+	secretData []byte) *corev1.Secret {
+	return createInternalSecret(secretNameAndNamespace,
+		storageClusterNameAndNamespace,
+		SourceLabel,
+		secretData)
+}
+
+// CreateDestinationSecret creates a GREEN secret
+func CreateDestinationSecret(secretNameAndNamespace types.NamespacedName,
+	storageClusterNameAndNamespace types.NamespacedName,
+	secretData []byte) *corev1.Secret {
+	return createInternalSecret(secretNameAndNamespace,
+		storageClusterNameAndNamespace,
+		DestinationLabel,
+		secretData)
+}
+
+//+kubebuilder:rbac:groups=multicluster.odf.openshift.io,resources=mirrorpeers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=multicluster.odf.openshift.io,resources=mirrorpeers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=multicluster.odf.openshift.io,resources=mirrorpeers/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=pods;secrets;configmaps;events,verbs=*
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=roles;rolebindings,verbs=*
+//+kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews,verbs=get;create
+//+kubebuilder:rbac:groups=certificates.k8s.io,resources=certificatesigningrequests;certificatesigningrequests/approval,verbs=get;list;watch;create;update
+//+kubebuilder:rbac:groups=certificates.k8s.io,resources=signers,verbs=approve
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=work.open-cluster-management.io,resources=manifestworks,verbs=*
+//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=managedclusteraddons/finalizers,verbs=*
+//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=managedclusteraddons;clustermanagementaddons,verbs=*
+//+kubebuilder:rbac:groups=addon.open-cluster-management.io,resources=managedclusteraddons/status,verbs=*
```

Comment:
We definitely don't need all of these here. Just add the required ones.

Reply:
Removed lines which are not relevant. Please take a look